libraries/render-utils/src/Model.cpp

```diff
@@ -282,11 +282,6 @@ void Model::setRenderItemsNeedUpdate() {
     emit requestRenderUpdate();
 }
 
-void Model::setPrimitiveMode(PrimitiveMode primitiveMode) {
-    _primitiveMode = primitiveMode;
-    setRenderItemsNeedUpdate();
```

Comment:
This was a major bug.  This wasn't checking if primitiveMode changed, so it would always call `setRenderItemsNeedUpdate`, which always calls `emit requestRenderUpdate()`.  As a result, all models were constantly trying to update themselves, which meant actual changes would take longer to process (because entity updates are timeboxed).  This meant that in a domain of say 500 models, even if most of them were static, the few dynamic entities would update much slower than expected.  Now, as you can see from the new stats, the number of total entities that need to be updated should be lower, which means more will be processed every frame.

Reply:
Great catch.